File: examples/governance_static_go/domain/assigned_role.go
package domain

import "time"

type AssignedRole struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	StakeholderId string `json:"stakeholder_id"`
	Role string `json:"role"`
	Name string `json:"name"`
	Email string `json:"email"`
	Team string `json:"team"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e AssignedRole) EventName() string { return "AssignedRole" }

func (e AssignedRole) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/identity_domain/lib/identity_domain/stakeholder/events/assigned_role.rb
module IdentityDomain
  class Stakeholder
    module Events
      class AssignedRole
        attr_reader :aggregate_id, :stakeholder_id, :role, :name, :email, :team, :status, :occurred_at

        def initialize(aggregate_id: nil, stakeholder_id: nil, role: nil, name: nil, email: nil, team: nil, status: nil)
          @aggregate_id = aggregate_id
          @stakeholder_id = stakeholder_id
          @role = role
          @name = name
          @email = email
          @team = team
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
